### DISCUSSION TITLE
matchPackagePrefixes instead of ignoreDeps

### DISCUSSION THREAD

**Discussion by fgblomqvist** [renovatebot/renovate#9474] [Request Help] [answered]
Instead of manually specifying a long list of packages in `ignoreDeps`, we decided to [follow the docs](https://docs.renovatebot.com/configuration-options/#ignoredeps) and add the following rules to our packageRules:
```json
{
  "packageRules": [
    {
      "matchPackagePrefixes": ["github.com/repo/with/packages/to/ignore/", "github.com/another/repo/to/ignore"],
      "enabled": false
    }
  ]
}
```

We added it at the top of our rules array, for visibility. After that rule, we just have a lot of `matchPaths` rules (monorepo). However, Renovate is not ignoring deps that start with the specified prefixes (this is in Golang btw).

Is there some hidden priority here that if you enable a certain path, it overrides any previous `matchPackagePrefixes` and simply deals with all deps that appear in that path? Before I dig into the code to find out I might as well ask since it might be useful to others as well.

Side note: I do find `matchPackagePrefixes` to be a bit ambiguous. According to the docs, it sounds like it targets dependency package names. However, it reads as if it targets our package names. E.g. "for all of your packages in your repo, if they have this prefix in their name, we will use this rule.". Maybe renaming it to `matchDependencyPrefix` or something similar would clear potential confusion?

**Comment by viceice** [comment, marked as answer]
package rules are all executed in order and all matching rules are applied

**Reply by rarkins** [reply]
Naming is consistent with packageNames and packagePatterns. And you can use packagePatterns instead as it's a regex and hence more powerful

**Reply by fgblomqvist** [reply]
Okay about the naming (sounds good).
However, maybe I'm missing something, but what about the priority? Are you saying that `matchPackagePatterns` takes higher priority than `matchPackagePrefixes` @rarkins?

@viceice Hmm so you're saying that the algo is something like this:
1. Found a dep x
2. Execute each packageRule (in the order they're defined) to get the desired properties
3. First it will match the `matchPackagePrefixes` rule and therefore get disabled/ignored
4. Then it will also match the `matchPaths` rule since it was found in one of those paths, and therefore get enabled again

In that case, it sounds like the problem we're having would be solved by moving the `matchPackagePrefixes` ignore rule to the end of the `packageRules` array.

**Reply by viceice** [reply]
yes, you should move that rule to last, if you like to force them always disabled.
